### DISCUSSION TITLE
Possibility of distribution of NC-AIO via Github Packages

### DISCUSSION THREAD

**Discussion by dienteperro** [nextcloud/all-in-one#3211] [Ideas] [answered]
Github Packages (https://docs.github.com/packages) can be used as an alternative to Docker containers public registry. Images are served right from Github, the public registry is at: https://ghcr.io

**Comment by szaimen** [comment]
Hi, what would be advantages over docker hub?

**Reply by Zoey2936** [reply]
It comes without the rate limit, if I remember correctly

**Reply by szaimen** [reply, marked as answer]
See https://github.com/nextcloud/all-in-one/discussions/2667 then

**Reply by dienteperro** [reply]
On top of the limit, there are software-export restrictions on Docker hub that limit developers from countries banned by US laws to use it at all (Docker Hub and any image hosted there). Github Container Registry is outside of those, at least by now. Other option is to put the images as a list, so developers from those countries can find an easy way to use alternative images editing the list instead of forking and keeping a total parallel repo just to change the images names. In my mind is that Nextcloud and AIO are community-backed projects, they should not be limited by one country restrictions.
A similar situation was faced by OpenBSD/OpenSSH in their early days.

**Reply by szaimen** [reply]
> Other option is to put the images as a list

You can find them here: https://github.com/nextcloud/all-in-one/blob/main/manual-install/latest.yml (search for image)

**Reply by KiyoNetcat** [reply]
Docker Hub is getting ratelimited to 10 pulls/hour for unauthenticated users in April: https://docs.docker.com/docker-hub/usage/#definition-of-limits (authenticated users with a free account get 100 pulls/hour)
Considering the AIO container pulls and deploys many other images, I'd like to see the images also distributed on ghcr.io or another container registry so that users without an account won't hit the limit when trying to update or set this up.

**Reply by szaimen** [reply]
Hi, lets track this in https://github.com/nextcloud/all-in-one/discussions/6066
